### PR TITLE
SYNCT-29: Logic for getting and persisting Audit logs.

### PR DESCRIPTION
**Pull request by pkornowski**

### CODE REVIEW

File: api/src/main/resources/liquibase.xml [hunk 1]
```diff
@@ -5,14 +5,20 @@
...
                   http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-1.9.xsd">
 
-    <changeSet id="sync2-2017-11-29" author="Soldevelo">
+    <changeSet id="sync2-2017-12-06" author="Soldevelo">
+        <preConditions onFail="MARK_RAN">
+            <not><tableExists tableName="sync_audit_message"/></not>
+        </preConditions>
         <comment>
-            Creating the sync_audit_message table
+            Creating the outgoingmessageexceptions_message table
```
Comment (ubap) [inline api/src/main/resources/liquibase.xml]: Table is named "sync_audit_message". I don't get the comment.

File: api/src/main/java/org/openmrs/module/sync2/api/impl/SyncAuditServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import org.openmrs.module.sync2.api.dao.SyncAuditDao;
+import org.openmrs.module.sync2.api.model.audit.AuditMessage;
+import org.openmrs.module.sync2.api.model.audit.AuditMessageList;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.List;
+
+public class SyncAuditServiceImpl extends BaseOpenmrsService implements SyncAuditService {
+
+    SyncAuditDao dao;
```
Comment (ubap) [inline api/src/main/java/org/openmrs/module/sync2/api/impl/SyncAuditServiceImpl.java]: Is default access modifier intentional?

File: api/src/main/java/org/openmrs/module/sync2/api/dao/impl/SyncAuditDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+
+@Repository
+public class SyncAuditDaoImpl implements SyncAuditDao {
+
+    @Autowired
+    DbSessionFactory sessionFactory;
```
Comment (ubap) [inline api/src/main/java/org/openmrs/module/sync2/api/dao/impl/SyncAuditDaoImpl.java]: Is default access modifier intentional?

File: omod/src/main/java/org/openmrs/module/sync2/web/controller/rest/SyncAuditRestController.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+
+@Controller("${artifactid}.SyncAuditRestController")
+@RequestMapping(value = "/rest/sync2", produces = MediaType.APPLICATION_JSON_VALUE)
+public class SyncAuditRestController {
+
+    private final Logger LOGGER = LoggerFactory.getLogger(SyncAuditRestController.class);
+
+    @Autowired
+    SyncAuditService syncAuditService;
```
Comment (ubap) [inline omod/src/main/java/org/openmrs/module/sync2/web/controller/rest/SyncAuditRestController.java]: Is default modifier intentional?

Comment (pkornowski) [reply omod/src/main/java/org/openmrs/module/sync2/web/controller/rest/SyncAuditRestController.java]: My oversight. Corrected.

File: api/src/main/java/org/openmrs/module/sync2/api/dao/impl/SyncAuditDaoImpl.java (hunk 3 above)
Comment (pkornowski) [reply api/src/main/java/org/openmrs/module/sync2/api/dao/impl/SyncAuditDaoImpl.java]: My oversight. Corrected.

File: api/src/main/java/org/openmrs/module/sync2/api/impl/SyncAuditServiceImpl.java (hunk 2 above)
Comment (pkornowski) [reply api/src/main/java/org/openmrs/module/sync2/api/impl/SyncAuditServiceImpl.java]: My oversight. Corrected.

File: api/src/main/resources/liquibase.xml (hunk 1 above)
Comment (pkornowski) [reply api/src/main/resources/liquibase.xml]: You're right. I used an existing pattern for this and I didn't notice that. Thanks.

**Pull request merged**